CiscoESA/cisco-esa/ingest/parser.yml

```diff
@@ -0,0 +1,381 @@
+          destination.ip: "{{parsed_event.message.dst}}"
+          event.timezone: "{{parsed_event.message.dtz}}"
+          destination.user.id: "{{parsed_event.message.duid}}"
+          destination.user.email: "{{parsed_event.message.duser}}"
+          destination.user.name: "{{parse_duser.message.duser_name}}"
+          destination.user.domain: "{{parse_duser.message.duser_domain}}"
+          server.ip: "{{parsed_event.message.dvc}}"
+          destination.mac: "{{parsed_event.message.dmac}}"
+          event.end: "{{parse_end_date.date}}"
+          file.inode: "{{parsed_event.message.fileId}}"
+          file.mtime: "{{parsed_event.message.get('fileModification Time')}}"
+          file.path:  "{{parsed_event.message.filePath}}"
+          file.type:  "{{parsed_event.message.fileType}}"
+          host.network.ingress.bytes: "{{parsed_event.message.in}}"
+          host.network.egress.bytes: "{{parsed_event.message.out}}"
+          event.outcome: "{{parsed_event.message.outcome}}"
+          event.action: "{{parsed_event.message.reason}}"
+          url.original: "{{parsed_event.message.request}}"
+          user_agent.original: "{{parsed_event.message.get('requestClient Application')}}"
+          http.request.referrer: "{{parsed_event.message.requestContext}}"
+          http.request.method: "{{parsed_event.message.requestMethod}}"
+          host.hostname: "{{parsed_event.message.shost or parsed_event.message.sourceHostName}}"
+          source.domain: "{{parsed_event.message.get('sourceDns Domain')}}"
+          source.mac: "{{parsed_event.message.smac}}"
+          source.port: "{{parsed_event.message.spt}}"
+          source.nat.ip: "{{parsed_event.message.get('source Translated Address')}}"
+          source.nat.port: "{{parsed_event.message.get('source TranslatedPort')}}"
+          event.start:  "{{parse_start_date.date}}"
+          email.message_id: "{{parsed_event.message.ExternalMsgID}}"
+          email.local_id: "{{parsed_event.message.ESAMID}}"
+          rule.name: "{{parsed_event.message.MailPolicy}}"
+          email.attachments: >
+            [
+              {% for filename, info in dict(json_event_esaattachmentdetails).items() %}
+                {
+                  "file": {
+                    "name": "{{filename}}",
+                    {% if info.get("AMP", {}).get("fileHash") is not none %}
+                      "hash": { "sha256": "{{info.AMP.fileHash}}" }
+                    {% endif %}
+                  }
+                }
+                {% if not loop.last %},{% endif %}
+              {% endfor %}
+            ]
+          url.full: >   
+            [
+              {% for url.full in dict(json_event_esaurldetails).values() %}
+                {
+                  "url": "{{url.full}}"{ % if url != dict(json_event_esaurldetails)|last % },{ % endif % }
+                }
+                { % if not loop.last % },{ % endif % }
+              { % endfor % }
+            ]
```

Comment:
As we can have several values, I think we should move this field in a custom one and have a list of urls as the value